### PR TITLE
Add branch protection

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Added Branch Protector CronJob
- Defined protection for `test-infra` repository
- Simplified plugins configuration

To test Branch Protector bot needs to has access to repository settings.

**Related issue(s)**

See #131

### CODE REVIEW

File: prow/plugins.yaml [hunk 1]
```diff
@@ -1,29 +1,10 @@
...
-  kyma-project/website:
-    - cat
-    - trigger
-  kyma-project/luigi:
+  kyma-project:
     - cat
     - trigger
+    - owners-label
+    - verify-owners
+    - approve
```
Comment (Demonsthere) [inline prow/plugins.yaml]: If we are modifying the plugin scope to be organisation wide, I'd just suggest adding the `welcome`, `lgtm` and `size` plugins to the list

Comment (michal-hudy) [reply prow/plugins.yaml]: I decided to not add any new plugin. I just change to scope to the organization level to make `plugins.yaml` file smaller.

**Comment by michal-hudy** [comment]
@aszecowka I decided to not add information about branch protection in the migration guide, because we ask people to set `skip_report: true`. In that case, the PR will be blocked, because it will be waiting for that status from Prow.

I think we should change this in follow up task when finalizing migration.

**Comment by aszecowka** [comment]
I think only missing part is documentation: it can be one, two sentences about branch protection.  BTW you can add information that we are using config_updater.

**Comment by aszecowka** [comment]
Could you add only one line why we need branch protection? @michal-hudy

**Pull request merged**